### ISSUE TITLE
Adding space at the end of a word, sometimes it automatcally goes to the beginning of the word!

### ISSUE THREAD

**Issue by sh-bd** [OpenBangla/OpenBangla-Keyboard#360]
Whenever I am writing something using phonetic layout and adding space at the end of a word, sometimes it (the space) automatically goes to the beginning of the word! 

**Please be aware** that I've switched from `ibus-avro` to `openbangla-keyboard` for this issue but it doesn't help either. I think there might be a problem in the `ibus`.

Screen record: [Screencast from 2023-12-12 20-49-26.webm](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/44128725/8f189d3b-e916-449a-8bae-8e5216770b07)

OS config: `Ubuntu 23.10 64-bit (Wayland)`

Do you know any solutions for this issue?
Thanks in advance.

**Comment by mominul** [comment]
@sh-bd are you using the 2.0.0 version?

**Comment by sh-bd** [comment]
@mominul I used the command `bash -c "$(wget -q https://raw.githubusercontent.com/OpenBangla/OpenBangla-Keyboard/master/tools/install.sh -O -)"` 

Isn't it for the latest version?

**Comment by mominul** [comment]
Yes, but there is a development version 3.0.0 which is on the [`develop`](https://github.com/OpenBangla/OpenBangla-Keyboard/tree/develop) branch. 

Does this bug happen on other applications too?

**Comment by sh-bd** [comment]
@mominul sorry, I couldn't catch that. What do you mean by other applications?

**Comment by mominul** [comment]
> @mominul sorry, I couldn't catch that. What do you mean by other applications?

Like the text editor that comes with Ubuntu, browser etc

**Comment by larina3315** [comment]
I have faced this issue with ibus OBK 2 on Fedora 39 with GNOME 45 using the .rpm file provided for Fedora 39 in the releases section.
I believe I did not face any similar issue on Fedora 38 with GNOME 44, it might have something to do with newer version of ibus dependencies and OBK.

> Does this bug happen on other applications too?

It happened everywhere for me, I could temporarily rebase to a Silverblue image and help with any debugging effort if required.

**Comment by mominul** [comment]
> It happened everywhere for me, I could temporarily rebase to a Silverblue image and help with any debugging effort if required.

@larina3315  Can you have a recheck with openbangla-ibus from the `develop` branch? If the issue doesn't occur then the bug is fixed by some improvements in the ibus engine.

**Comment by larina3315** [comment]
> > It happened everywhere for me, I could temporarily rebase to a Silverblue image and help with any debugging effort if required.
> 
> @larina3315 Can you have a recheck with openbangla-ibus from the `develop` branch? If the issue doesn't occur then the bug is fixed by some improvements in the ibus engine.

Sadly the issue still appears to occur, Sometimes one or two extra spaces will get added at the beginning of a word, sometimes two words that should have a space between them that the user inserted, don't have a space at that location.

It only appears for me when I type fast, it seems like the faster I type, the likelihood of the bug occuring increases.

**Comment by larina3315** [comment]
[Screencast from 2023-12-26 22-47-55.webm](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/725b226f-e6f4-4392-bd59-7df96f901947)

**Comment by larina3315** [comment]
> Screencast.from.2023-12-26.22-47-55.webm

note that I have typed a space after each word, so there is no manual error, this bug also doesn't occur on fcitx OBK, only on ibus OBK

**Comment by larina3315** [comment]
> Screencast.from.2023-12-26.22-47-55.webm

note that there are extra spaces inserted after পরশুদিন (1st line), এত (2nd line), দিলামতুমি (2nd line).
It might be unclear at first glace because of the font not being monospace.

**Comment by mominul** [comment]
It'd be tricky to reproduce, I'll have a go at it when my Linux PC gets into a working state :cry:

**Comment by larina3315** [comment]
Understandable bhai, thank you for the amount of effort you have put into OBK.

**Comment by larina3315** [comment]
Just for visibility, as mentioned in https://github.com/OpenBangla/OpenBangla-Keyboard/discussions/367
fcitx OBK 3 also shows similar buggy behaviour on Debian 11 KDE when using the packages provided in the discussion, but they work fine on Linux Mint 21 Cinnamon.

[Screencast from 2024-01-08 01-03-18.webm](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/faf5d24c-f315-4d0a-85a8-d327846bc194)

**Comment by larina3315** [comment]
This bug can be reliably reproduced everytime on the Lapce 0.3.1 Flathub package while using Fedora 39 and ibus obk3

**Comment by mominul** [comment]
I have filled an issue ibus/ibus#2601
